xwiki-platform-core/xwiki-platform-realtime/xwiki-platform-realtime-wysiwyg/xwiki-platform-realtime-wysiwyg-webjar/src/main/webjar/wysiwygEditor.js

```diff
@@ -525,6 +526,68 @@ define('xwiki-realtime-wysiwygEditor', [
 
           realtimeOptions.onLocal();
           createSaver(info);
+
+          // We need to prevent the diff modal from reloading the editor
+          // because when reloading the realtime editor when editing with
+          // another user, the content will not be loaded from the saved
+          // version, but from the existing realtime session.
+
+          // Instead of reloading the editor, the content of the realtime
+          // session should be replaced with the currently saved version.
+          $(document).on('xwiki:dom:updated', (event, param) => {
+            // The preview diff modal triggers a xwiki:dom:updated event
+            // once initialized (opened), after registering the event handlers
+            if (param?.elements?.first()?.id === "previewDiffModal") {
+
+              var submitButton = $('#submitDiffButton');
+
+              // We want to keep the existing behavior when reload is
+              // not the selected option.
+              // Passing true to the clone method allows to copy event handlers,
+              // allowing us to call the old event handler even after changing the
+              // one on the original button.
+              var oldSubmit = submitButton.clone(true);
+
+              // Replace the click event handler of the submit button.
+              submitButton.off("click");
+              submitButton.on("click", () => {
+                if ($('input[name=warningConflictAction]:checked').val() === "reload") {
+                  // Reload is selected so we get the currently saved version from the service.
+                  $.ajax({
+                    url: new XWiki.Document('Ajax', 'RTFrontend').getURL('get', 'xpage=plain&outputSyntax=plain'),
+                    method: 'POST',
+                    dataType: 'json',
+                    data: {
+                      wiki: editor.config.sourceDocument.wiki,
+                      document: XWiki.Model.serialize(doc.documentReference),
```

Comment:
```suggestion
                      sheet: 'RTFrontend.Ajax',
```

I'm not sure why we need to pass the wiki. I think we can drop it.

Reply:
I removed the Ajax service. Getting rid of these issues.
Thanks.